Sources/Configuration/Configuration.swift

```diff
@@ -25,4 +25,13 @@ public protocol Configuration {
     ///
     /// - return: The request headers.
     func headers(for request: Request) -> RequestHeaders?
+    /// Request queries that should be applied to all requests
+    var defaultQuery: RequestQuery? { get }
```

Comment:
I would align this with the `headers` function.

```swift
func query(for request: Request) -> RequestQuery?
```

This way you can make the query specific, it could come in useful when you want to ignore the default query for a certain request.

Reply:
I see what you mean there, but since the `Requests` themselves already have a `query` property, it felt a bit redundant to me.

But I can change it to make it inline with the way the `headers` are handled. 

Another thought was to add something similar like the `Interceptor`, which would return the final `URLRequest` right before sending it out, so any additional configuration could be done there. But this felt like a more impactful change than this one.

Reply:
@fousa I've reverted my changes, and tried a different approach with an injectable `RequestBuilder`, which felt more flexible to me. 

Let me know what you think ;)